### DISCUSSION TITLE
Getting battlesnake to run in Google Colab?

### DISCUSSION THREAD

**Discussion by daveSMUS** [BattlesnakeOfficial/feedback#250] [Q&A]
With Repl changing (removing) its education plan we are looking at options for our students to keep battling. We already are using Google's Colab for other course work and so I tried to get it up and running in there:

https://colab.research.google.com/drive/1cOc2f6dUGKfJoceKzzKqbtDUqyHcCPmi?usp=sharing

When I execute, It looks like the server is running properly. When I check the URL in the browser it get this:

{"apiversion":"1","author":"dave","color":"#FF0000","head":"default","tail":"default"}

But when I try to add that URL to my snake I am getting 403 Forbidden errors. Any idea on what I am doing wrong or what to try next? 

Thank you!

**Comment by ItsGoldeneyes** [comment]
Testing with your notebook, I'm not getting a 403 error while I have the notebook open, but when I close it the deploy link 403's.Colab is not meant for hosting webservers (like snakes), so it won't work well for your usecase. You'd likely have to keep the notebook open while the snake runs, which is less than ideal.

I would reach out to the folks at [Railway.app](https://railway.app/) and ask about an education plan. Create a trial account, link your Github and reach out to billing@railway.app. I've been pushing the team to kick off an education team for a while now and have seen them grant exceptions before. It's worth a shot.

If they decline, Railway itsself is very reasonably priced. Personally, I run 3 Python snakes on there, running me ~50 cents/month/snake. If you upgrade to the Hobby plan, you can host as many as you like. If you have any issues with this, reach out to me at a_dumb__ on their discord, I'm a helper there.

Repl closing down their education plan really, really sucks. I'm really sorry to hear that. The trend of hosting services removing their free plans is only accelerating, I'm not a fan.

**Reply by daveSMUS** [reply]
Thanks! We'd be ok with leaving the notebook open when using the snake... Repl often needed a 'poke' anyways. 

But I can't get around the 403 in the Battlesnake load page. I think I have figured out that a ping doesn't work on a Colab and it looks like that is how Battlesnake site is verifying a working snake? I wonder if that can be tweaked.

I will look at Railway for sure. Thanks. GitHub Codespace is another place I am looking. I got that one working, but it is a bit messy to launch. Colab is so beautifully shareable with students.

**Reply by daveSMUS** [reply]
Nope. It is an authentication issue. When I browse in another profile I can't get to the URL either. Looks like colab is keeping the HTTP service tied to the user.

**Reply by ItsGoldeneyes** [reply]
Good call, I tried opening the URL in an incognito window and got a 403. Unfortunate
